### PR TITLE
Allow pickle protocol to be overridden

### PR DESCRIPTION
**Pull request by jakirkham**
This provides the option to override the pickle protocol used. The default behavior is still to use the `HIGHEST_PROTOCOL`. However this provides users an easy way to customize the pickle protocol through the `distributed.yaml` config.

cc @mrocklin @quasiben

### CODE REVIEW

File: distributed/distributed.yaml [hunk 1]
```diff
@@ -170,6 +170,8 @@ distributed:
     log-length: 10000  # default length of logs to keep in memory
     log-format: '%(name)s - %(levelname)s - %(message)s'
     pdb-on-err: False       # enter debug mode on scheduling error
+pickle:
+  protocol: null  # specify the pickle protocol to use
```
Comment (jakirkham) [inline distributed/distributed.yaml:174]: Just placed this at the global level as it is not really worker or scheduler specific. Not picky on where it lives though.

**Comment by mrocklin** [comment]
Interesting.  I'm curious, what was the motivation?

**Comment by jakirkham** [comment]
Issue ( https://github.com/dask/distributed/issues/4011 ) encouraged me to put it in a PR. Albeit that issue is discussing a different solution. Could also be useful for debugging/helping users.

**Comment by mrocklin** [comment]
Thanks for the speedy response to that issue.  If we were to go this approach (which seems sensible regardless) I would then ask the question

*what should we make the default?  4 or 5?*

I suspect that your answer is likely "5, because it's faster" while I probably tend towards "4, because it's robust".  I'm curious, did we ever speed test pickle protocol 5 on a real-world workload?  Would something like `x + x.T` be made considerably faster by it?  Given the cost around Python version mismatches it might be worth verifying that the performance benefit is felt in real-world situations

**Comment by jakirkham** [comment]
Closing since this seems to be addressed by PR ( https://github.com/dask/distributed/pull/4019 ).

**Comment by mrocklin** [comment]
Woot.  Thanks for the early effort @jakirkham

**Pull request closed without merging**